src/api/entities/CorporateAction/index.ts

```diff
@@ -99,4 +107,85 @@ export class CorporateAction extends Entity<UniqueIdentifiers> {
     this.defaultTaxWithholding = defaultTaxWithholding;
     this.taxWithholdings = taxWithholdings;
   }
+
+  /**
+   * Retrieve whether the Corporate Action exists
+   */
+  public async exists(): Promise<boolean> {
+    const { context, id, ticker } = this;
+
+    const corporateAction = await context.polymeshApi.query.corporateAction.corporateActions(
+      stringToTicker(ticker, context),
+      numberToU32(id, context)
+    );
+
+    return corporateAction.isSome;
+  }
+
+  /**
+   * Retrieve the Checkpoint associated with this Corporate Action. If the Checkpoint is scheduled and has
+   *   not been created yet, the corresponding CheckpointSchedule is returned instead. A null value means
+   *   the Corporate Action was created without an associated Checkpoint
+   */
+  public async checkpoint(): Promise<Checkpoint | CheckpointSchedule | null> {
+    const {
+      context: {
+        polymeshApi: { query },
+      },
+      context,
+      id,
+      ticker,
+    } = this;
+
+    const rawTicker = stringToTicker(ticker, context);
+
+    const corporateAction = await query.corporateAction.corporateActions(
+      rawTicker,
+      numberToU32(id, context)
+    );
+
+    if (corporateAction.isNone) {
```

Comment:
Should this be calling `exists` instead?

Reply:
Exists calls `corporateAction.corporateActions`, so we would be calling the same endpoint twice. I can refactor that call to avoid a bit of repetition